uxsim/uxsim.py

```diff
@@ -169,10 +169,13 @@ def generate(s):
                         outlinks = sorted(set(outlinks) - set(veh.links_avoid))
                     
                     preference = np.array([veh.route_pref[l.id] for l in outlinks], dtype=float)
-                    if sum(preference) > 0:
-                        outlink = s.W.rng.choice(outlinks, p=preference/sum(preference))
+                    if s.W.hard_deterministic_mode == False:
```

Comment:
```suggestion
                    if not s.W.hard_deterministic_mode:
```
would be more Pythonic